src/engine/LoadURL.cpp

```diff
@@ -0,0 +1,167 @@
+Result LoadURL::computeResult(bool requestLaziness) {
+  try {
+    return computeResultImpl(requestLaziness);
+  } catch (const ad_utility::CancellationException&) {
+    throw;
+  } catch (const ad_utility::detail::AllocationExceedsLimitException&) {
+    throw;
+  } catch (const std::exception&) {
+    // If the `SILENT` keyword is set, catch the error and return a neutral
+    // Element.
+    if (loadURLClause_.silent_) {
+      IdTable idTable{getResultWidth(), getExecutionContext()->getAllocator()};
+      idTable.emplace_back();
+      for (size_t colIdx = 0; colIdx < getResultWidth(); ++colIdx) {
+        idTable(0, colIdx) = Id::makeUndefined();
+      }
+      return {std::move(idTable), resultSortedOn(), LocalVocab{}};
+    }
+    throw;
+  }
+}
+
+// _____________________________________________________________________________
+Result LoadURL::computeResultImpl(bool) {
+  // TODO<qup42> implement lazy loading; requires modifications to the parser
+  LOG(INFO) << "Loading RDF dataset from " << loadURLClause_.url_.asString()
+            << std::endl;
+  HttpOrHttpsResponse response =
+      getResultFunction_(loadURLClause_.url_, cancellationHandle_,
+                         boost::beast::http::verb::get, "", "", "");
+
+  auto throwErrorWithContext = [this, &response](std::string_view sv) {
+    this->throwErrorWithContext(sv, std::move(response).readResponseHead(100));
+  };
+
+  if (response.status_ != boost::beast::http::status::ok) {
+    throwErrorWithContext(absl::StrCat(
+        "RDF dataset responded with HTTP status code: ",
+        static_cast<int>(response.status_), ", ",
+        toStd(boost::beast::http::obsolete_reason(response.status_))));
+  }
+  if (response.contentType_.empty()) {
+    throwErrorWithContext(
+        "QLever requires the `Content-Type` header to be set for LoadURL.");
+  }
+  // If the `Content-Type` is not one of the media types known to QLever, then
+  // std::nullopt is returned by `toMediaType`.
+  std::optional<ad_utility::MediaType> mediaType =
+      ad_utility::toMediaType(response.contentType_);
+  if (!mediaType) {
+    throwErrorWithContext(
+        absl::StrCat("Unknown `Content-Type` \"", response.contentType_, "\""));
+  }
+  if (!ad_utility::contains(SUPPORTED_MEDIATYPES, mediaType.value())) {
+    throwErrorWithContext(absl::StrCat(
+        "Unsupported value for `Content-Type` \"", toString(mediaType.value()),
+        "\". Supported are \"text/turtle\" and "
+        "\"application/n-triples\"."));
+  }
+  using Re2Parser = RdfStringParser<TurtleParser<Tokenizer>>;
+  auto parser = Re2Parser();
+  std::string body;
+  for (const auto& bytes : response.body_) {
+    body.append(reinterpret_cast<const char*>(bytes.data()), bytes.size());
+  }
+  parser.setInputStream(body);
+  LocalVocab lv;
+  IdTable result{getResultWidth(), getExecutionContext()->getAllocator()};
+  size_t row_idx = 0;
+  for (auto& triple : parser.parseAndReturnAllTriples()) {
```

Comment:
Does this interface batch in some way, or does it simply return you a plain vector?
Because then this also is a possible source of hidden out of memory issues.
Let me know, how easy this would be to implement for you in a batched way.